machine_learning/data_downloader.py

```diff
@@ -0,0 +1,167 @@
+        os.mkdir(dirname)
+
+    download_log_handler = open(str(dirname) + '.txt', 'w')
+    filename = os.path.join(os.getcwd(), filename)
+    df = pd.read_csv(filename, header=0, error_bad_lines=False)
+    new_dir = os.path.join(path, dirname)
+    os.chdir(new_dir)
+    download_log_handler.write(time.asctime(time.localtime(time.time())) + '\n')
+    urls = df['Last Two - Side by Side']
+
+    for index in range(0, len(df['Last Two - Side by Side'])):
+        from_version_uri, to_version_uri, ids = get_storage_uris(urls[index])
+        if (from_version_uri == None or to_version_uri == None):
+            download_log_handler.write('Incorrect urls' + '\n')
+            continue
+
+        from_version_response = requests.get(from_version_uri)
+        to_version_response = requests.get(to_version_uri)
+        from_filename = str(ids['site_id'] + '_' + ids['page_id']
+                            + '_' + ids['from_version_id'])
+        to_filename = str(ids['site_id'] + '_' + ids['page_id']
+                          + '_' + ids['to_version_id'])
+
+        if (from_version_response.ok and to_version_response.ok):
+            with open('change_' + str(index + 1) + '_' + from_filename
+                      + '_from_version.html', 'w', encoding = 'utf-8') as f:
+                f.write(from_version_response.text)
+            with open('change_' + str(index + 1) + '_' + to_filename
+                      + '_to_version.html', 'w', encoding = 'utf-8') as f:
+                f.write(to_version_response.text)
+            download_log_handler.write('Change ' + str(index + 1)
+                                       + ' ' + ' '.join(from_filename.split('_'))
+                                       + ' from version uri : '
+                                       + from_version_uri + '\n')
+            download_log_handler.write('Change ' + str(index + 1)
+                                       + ' ' + ' '.join(to_filename.split('_'))
+                                       + ' to version uri : '
+                                       + to_version_uri + '\n')
+        else:
+            if (not from_version_response.ok):
+                download_log_handler.write('Change ' + str(index + 1)
+                                           + ' '.join(from_filename.split('_'))
+                                           + ' from version issue :'
+                                           + from_version_response.text)
+            if (not to_version_response.ok):
+                download_log_handler.write('Change ' + str(index + 1)
+                                           + ' '.join(to_filename.split('_'))
+                                           + ' to version issue :'
+                                           + to_version_response.text)
+            continue
+
+    download_log_handler.close()
+    os.chdir('..')
+    return True
+
+def main():
+
+    parser = argparse.ArgumentParser(description='Run downloader script')
```

Comment:
Can we please stick to one way of doing CLI arg parsing? Existing code uses DocOpt. For example: https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/master/web_monitoring/cli.py#L103-L114

Unless we've decided we don't want to use it at all anymore.

Reply:
I've made the changes to DocOpt. I think it's better if we stick with it.